corelib-common-storage/src/main/java/eu/europeana/corelib/storage/impl/MongoProviderImpl.java

```diff
@@ -21,22 +21,23 @@
 
     private static final Logger LOG                   = LogManager.getLogger(MongoProviderImpl.class);
 
-    private static final int MAX_CONNECTION_IDLE_MILLIS   = 30000;
-
     private MongoClient mongo;
     private String definedDatabase;
 
     /**
-     * Create a new MongoClient based on a connectionUrl, e.g.
+     * Create a new MongoClient based on a connectionUrl with MongoClientOptions e.g.
      * mongodb://user:password@mongo1.eanadev.org:27000/europeana_1?replicaSet=europeana
      * @see <a href="http://api.mongodb.com/java/current/com/mongodb/MongoClientURI.html">
      *     MongoClientURI documentation</a>
      *
      * @param connectionUrl
+     * @param maxConnectionIdleTime sets the maxConnectionIdleTime if not empty
      */
-    public MongoProviderImpl(String connectionUrl) {
+    public MongoProviderImpl(String connectionUrl, String maxConnectionIdleTime) {
```

Comment:
Did you check if Metis is using the old constructor?
Since you changed the signature of this class, this could result in Metis not being able to upgrade to a new version of Corelib without having to also provide a maxConnectionIdleTime parameter. If that's the case they then need to research what it means and what proper values are. A more user-friendly implementation would be to keep the old MongoProviderImpl and add a second constructor with with the maxConnectionIdleTime

Reply:
Ohh yes, I would better create a second constructor with the maxConnectionIdleTime